packages/grafana-toolkit/src/cli/utils/useSpinner.ts

```diff
@@ -1,20 +1,16 @@
-import ora = require('ora');
+import ora from 'ora';
 
-type FnToSpin<T> = (options: T) => Promise<any>;
-
-export function useSpinner<T = void>(spinnerLabel: string, fn: FnToSpin<T>, killProcess = true) {
-  return async (options: T) => {
-    const spinner = ora(spinnerLabel);
-    spinner.start();
-    try {
-      await fn(options);
-      spinner.succeed();
-    } catch (e) {
-      console.trace(e); // eslint-disable-line no-console
-      spinner.fail(e.message || e);
-      if (killProcess) {
-        process.exit(1);
-      }
+export const useSpinner = async (label: string, fn: () => Promise<any>, killProcess = true) => {
```

Comment:
when do you export const vs export function?  

I know the const stuff is useful for binding `this` but any use here?

Reply:
TypeScript generics don't support arrow functions, so I use them then as well. I give preference to arrow functions mostly.